AmazonSQS/AmazonSQSJCAAPI/src/main/java/fish/payara/cloud/connectors/amazonsqs/api/inbound/SQSPoller.java

```diff
@@ -40,38 +40,59 @@
 package fish.payara.cloud.connectors.amazonsqs.api.inbound;
 
 import fish.payara.cloud.connectors.amazonsqs.api.OnSQSMessage;
+import jakarta.json.Json;
+import jakarta.json.JsonArray;
+import jakarta.json.JsonException;
+import jakarta.json.JsonObject;
+import jakarta.json.JsonReader;
+import jakarta.json.JsonValue;
 import jakarta.resource.spi.BootstrapContext;
 import jakarta.resource.spi.endpoint.MessageEndpointFactory;
 import jakarta.resource.spi.work.WorkException;
+import java.io.IOException;
+import java.io.StringReader;
 import java.lang.reflect.Method;
 import java.util.Arrays;
 import java.util.List;
 import java.util.TimerTask;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 import java.util.stream.Collectors;
+import software.amazon.awssdk.core.ResponseInputStream;
 import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
 import software.amazon.awssdk.services.sqs.SqsClient;
 import software.amazon.awssdk.services.sqs.model.Message;
 import software.amazon.awssdk.services.sqs.model.QueueAttributeName;
 import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import software.amazon.awssdk.utils.IoUtils;
 
 /**
  * @author Steve Millidge (Payara Foundation)
  */
 class SQSPoller extends TimerTask {
 
-    AmazonSQSActivationSpec spec;
-    BootstrapContext ctx;
-    MessageEndpointFactory factory;
-    SqsClient client;
+    private final AmazonSQSActivationSpec spec;
+    private final BootstrapContext ctx;
+    private final MessageEndpointFactory factory;
+    private final SqsClient client;
+    private S3Client s3;
+    private static final String S3_BUCKET_NAME_KEY = "s3BucketName";
+    private static final String S3_KEY_KEY = "s3Key";
```

Comment:
same as the other PR,  probably just with S3_KEY name of variable:
```suggestion
    private static final String S3_KEY = "s3Key";
```